nodejs/week2/meal-sharing/src/backend/api/reservations-router.js

```
@@ -0,0 +1,19 @@
+const express = require("express");
+const router = express.Router();
+
+const reservations = require("./../data/reservations.json");
+
+// Respond with the json for all reservations *
+router.get("/", async (request, response) => {
+  response.send(reservations);
+});
+
+//Respond with the json for the reservation with the corresponding id
+router.get("/:id", async (request, response) => {
+  const reservationId = request.params.id;
+  response.send(
+    reservations.filter((reservation) => reservation.id == reservationId)
```

You should always check for the typeOf for id, with parseInt or Number or isNaN or etc.
If the user entered a string instead of a number accidentally, the code will not respond with anything. so the query should respond with data only when the correct input is provided, else it should throw an error saying wrong input.